### ISSUE TITLE
HLIL optimises out stack stores, even when pointers to the stack escape

### ISSUE THREAD

**Issue by dougallj** [Vector35/binaryninja-api#1728]
Using the following C code (compiled with `clang -O3`), the HLIL doesn't show the `1000` assignment.
```
struct params {
	int f1;
	int f2;
};

int func(void (*cb)(struct params *), int a2) {
	struct params p;
	p.f1 = a2;
	p.f2 = 1000;
	cb(&p);
}
```
This can be rectified by changing the type of `var_10` to `int[2]`, but there's nothing visible to indicate you should do that unless you check `MLIL` or lower.

<img width="1212" alt="de0" src="https://user-images.githubusercontent.com/615170/83130314-778ddd00-a121-11ea-819b-206cfd20faf3.png">

**Comment by yrp604** [comment]
Found this issue while looking for another bug, this is fixed now:

```
func:
   0 @ 00401110  int64_t rax
   1 @ 00401110  int64_t var_8 = rax
   2 @ 00401114  var_8.d = arg2
   3 @ 00401117  var_8:4.d = 1000
   4 @ 00401125  return arg1(&var_8)
```

**Comment by plafosse** [comment]
This is fixed as of 2.3
